### PR TITLE
Optimise follow-based recommendations

### PR DESCRIPTION
**Pull request by laurieboyes**
Significantly speeds up the query, whilst not having too great a negative effect on the results themselves (but worth keeping an eye on).

~~The degree of the speed increase seems to vary from user to user but my informal testing has shown it can make it anywhere between 2 and 10 times faster~~
The performance improvement figures given above are likely to have appeared more dramatic due to the effects Neo4j's internal caching (lol). A more reliable stat is to say that it reduces the number of db hits by a factor of around 4.5 (although this factor will increase along with our number of myFT users / follows)

Tried it for a number of users on the new cluster and it didn't break anything 👍

### CODE REVIEW

**Comment by keirog** [comment]
😮  👍

**Comment by wheresrhys** [comment]
😮 👍

File: server/v2/controllers/recommendation/user-concepts.js [hunk 1]
```diff
@@ -4,7 +4,8 @@ const db = require('../../../utils/db').read;
 const popularConcepts = require('../../lib/most-popular-concepts');
 /**
  * Current logic:
- * Get users whom follow same concepts as you
+ * Get users who follow the same concepts as you
```
Comment (phamann) [inline server/v2/controllers/recommendation/user-concepts.js]: Pretty sure that was a correct use of "whom" 😜

Comment (wheresrhys) [inline server/v2/controllers/recommendation/user-concepts.js]: Whom says?

Comment (keirog) [inline server/v2/controllers/recommendation/user-concepts.js]: Laurie’s right…

I find this technique very helpful and use it pretty much every day: change the sentence into a question and try to answer it with either “they” or “them”. If “the**m**” makes sense in that context and not “they”, then it should be “who**m**”.

“Who follows?” -> “They follow”. “Them follows” doesn’t make sense.
An example of “whom” would be:
“Whom should I follow?”, where we could change that to “Follow them” and “Follow they” doesn’t make sense.

**Comment by phamann** [comment]
I'm happy for this to be shipped 👍

**Comment by phamann** [comment]
Can we also increase the cache time and `surrogate-key` whilst we're at it :)

**Comment by laurieboyes** [comment]
Cheers 👍 Yep I can look at that next Thursday aft

**Pull request merged**